code-reviews/recipes/Terraform.md

```diff
@@ -53,10 +65,19 @@ echo -e "\n\n>>> Terraform validate"
 terraform validate
 ```
 
-> TODO: Add example of how to use this in an AzDO build validation pipeline
-
 ## Code Review Checklist
 
-In addition to the [Code Review Checklist](../readme.md) you should also look for these terraform specific code review items
-
-> TODO: Add checklist
+In addition to the [Code Review Checklist](../readme.md) you should also look for these Terraform specific code review items
+
+* [ ] Are all providers used in the terraform scripts [versioned](https://www.terraform.io/docs/configuration/providers.html#provider-versions) to prevent breaking changes in the future?
+* [ ] Are modules split into separate .tf files where appropriate?
+* [ ] Does the repository contain a `README.md` describing the architecture provisioned?
+* [ ] Is the Terraform project configured using Azure Storage as remote state backend?
+* [ ] Is the remote state backend storage account key is stored a secure location (e.g. Azure Key Vault)?
+* [ ] If Terraform code is mixed with application source code, is the Terraform code isolated into a dedicated folder?
+* [ ] If the infrastructure is going to different depending on the environment (e.g. Dev, UAT, Production), are the environment specific parameters are supplied via .tfvars file?
+* [ ] Are the resource definitions and data sources handled correctly in the Terraform scripts?
+    resource : Indicates to Terraform that the current configuration is in charge of managing the life cycle of the object
```

Comment:
Nit: Consider sub bullet points for resource and data descriptions

Reply:
@tompaana - The sub-bullet points making the "Read me" lint check to fail, and that is the reason I had to add them in the same line.